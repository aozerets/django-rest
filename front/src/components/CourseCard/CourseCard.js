import React from "react";
import './CourseCard.scss';
import '../../main.scss';
import { toggleSignCourse } from "../../actions";
import { connect } from "react-redux";

export class CourseCard extends React.Component {
  constructor(props) {
    super(props);
  }
  
  highlightOption = (ev) => {
    if (ev.type === 'mouseenter') {
      ev.currentTarget.classList.add('highlighted');
    } else {
      ev.currentTarget.classList.remove('highlighted');
    }
  };
  
  render() {
    const {toggleSignCourse, name, title, started, more} = this.props;
    return (
      <div className={`course-card course-card__${name} animate-card-in option`}
           onMouseEnter={this.highlightOption}
           onMouseLeave={this.highlightOption}>
        <h2 className="course-card__title">{title}</h2>
        <h4 className="course-card__more-info">Starting {started}!!!!</h4>
        <p className="course-card__more-info">Some more info about this awesome course you are considering!: {more}</p>
        <p className="course-card__action"><a className="course-card__btn" onClick={toggleSignCourse}>Sign it</a></p>
        <p className="course-card__action"><a className="course-card__btn" onClick={this.goToLessons}>Lessons</a></p>
      </div>
    )
  }
}
const mapDispatchToProps = dispatch => {
  return {
    toggleSignCourse: () => dispatch(toggleSignCourse())
  };
};
const CourseCardContainer = connect(null, mapDispatchToProps)(CourseCard);

export default CourseCardContainer;
