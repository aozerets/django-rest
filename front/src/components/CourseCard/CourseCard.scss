.course-card {
  animation-delay: .8s;
  overflow: hidden;
  margin: 1em 1em;
  border: 1px;
  border-radius: .8rem;
  width: 316px;
  height: 316px;
  box-shadow: 0 0 8px 1px #F5F5F5;
  position: relative;
  opacity: 0;
  &__title {
    animation-delay: .6s;
    text-transform: uppercase;
    margin-bottom: 5rem;
    font-family: "Roboto", "Lucida Grande", "DejaVu Sans", "Bitstream Vera Sans", Verdana, Arial, sans-serif;
    color: #F5F5F5;
  }
  &__action {
    display: inline-block;
    margin: 1em .5em;
    -webkit-transform: scale(.2);
    transform: scale(.2);
    visibility: hidden;
    color: #000;
  }
  &__more-info {
    visibility: hidden;
  }
  &__python {
    background: url(../../../src/public/image/python-logo-dark.jpg) no-repeat 100%;
    background-size: cover;
    padding: 0;
  }
  &__js {
    background: url(../../../src/public/image/js-logo.jpg) no-repeat 100%;
    background-size: cover;
    padding: 0;
  }
  &__kotlin {
    background: url(../../../src/public/image/kotlin-logo.jpg) no-repeat 100%;
    background-size: cover;
    padding: 0;
  }
  &__btn {
    justify-content: center;
    cursor: pointer;
    background: #C864FF;
    border-radius: 1.4rem;
    color: #fff;
    display: flex;
    padding: 0;
    margin: -2em auto;
    text-align: center;
    width: 10rem;
    &:hover {
      text-decoration: none;
      background-color: darken(#C864FF, 30%);
      //color: #000;
    }
  }
}
.option {
  display: inline-block;
  -webkit-transform: scale(.2);
  transform: scale(.2);
  color: #000;
  -webkit-transition: background .2s ease-out,
  -webkit-transform .5s cubic-bezier(0, 1.7, .3, 1);
  transition: transform 1.5s cubic-bezier(0, 1.7, .3, 1),
  background .2s ease-out,
  -webkit-transform .5s cubic-bezier(0, 1.7, .3, 1);

  &.highlighted {
    background: #fff;
    border-color: #eee;
    background-size: cover;
    -webkit-transform: scale(1.2);
    transform: scale(1.2);

    .course-card__title {
      color: #000;
      -webkit-transform: translateY(-1em) scale(1.2);
      transform: translateY(-.5em) scale(1.2);
    }

    .course-card__more-info {
      -webkit-transform: translateY(-1em);
      transform: translateY(-1em);
      visibility: visible;
    }

    .course-card__action {
      -webkit-transform: translateY(2em);
      transform: translateY(2em);
      visibility: visible;
    }
  }

  & > * {
    -webkit-transition: -webkit-transform .5s cubic-bezier(0, 1, .3, 1);
    transition: -webkit-transform .5s cubic-bezier(0, 1, .3, 1);
    transition: transform .5s cubic-bezier(0, 1, .3, 1);
  }
}
@keyframes fade-slide-down {
  0% {
    opacity: 0;
    transform: translateY(-4rem);
  }
  100% {
    opacity: 1;
    transform: none;
  }
}