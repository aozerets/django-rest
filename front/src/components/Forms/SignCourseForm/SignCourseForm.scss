.modal {
  display: none;
  position: fixed;
  z-index: 1;
  padding-top: 150px;
  left: 0;
  top: 0;
  width: 100%;
  height: 100%;
  overflow: auto;
  background-color: rgba(0,0,0,0.4);
  &__show {
    display: block;
    position: absolute;
    z-index: 1;
    padding-top: 150px;
    left: 0;
    top: 0;
    width: 100%;
    height: 100%;
    overflow: auto;
    background-color: rgba(0,0,0,0.4);
  }
  &__content {
    position: relative;
    background-color: #F5F5F5;
    margin: auto;
    padding: 0;
    border: 1px solid #888;
    width: 40%;
    box-shadow: 0 4px 8px 0 rgba(0,0,0,0.2),0 6px 20px 0 rgba(0,0,0,0.19);
    animation: modal-top .6s cubic-bezier(0, 0.9, 0.3, 1.2) forwards;
  }
  &__close {
    color: #F5F5F5;
    float: right;
    font-size: 28px;
    font-weight: bold;
  }
  &__close:hover {
    color: #000;
    text-decoration: none;
    cursor: pointer;
  }
  &__title {
    margin: 0 auto;
  }
  &__header {
    margin: 0 auto;
    padding: 1rem 1rem;
    background-color: #C864FF;
    color: #F5F5F5;
  }
  &__body {
    margin: 0 auto;
    padding: 1rem 1rem;
  }
  &__footer {
    margin: 0 auto;
    padding: 1rem 1rem;
    background-color: #C864FF;
    color: #F5F5F5;
  }
}

@keyframes modal-top {
  0% {
    top: -300px;
    opacity: 0;
    transform: translateY(-4rem);
  }
  100% {
    top: 0;
    opacity: 1;
  }
}
