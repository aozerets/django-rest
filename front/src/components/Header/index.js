import React from 'react';
import { connect } from 'react-redux'

import HeaderTitleContainer from '../HeaderTitle';
import TopCourses from '../TopCourses';
import ProfileContainer from '../Profile';
import News from '../News';
import './Header.scss';
import '../../main.scss';
import LoginContainer from "../Login";
import RegistrationContainer from "../Registration";

class Header extends React.Component {
  constructor(props) {
    super(props);
  }
  
  renderSwitch = name => {
    console.log('switched');
    switch (name) {
      case 'profile':
        return <ProfileContainer />;
      case 'login':
        return <LoginContainer />;
      case 'registration':
        return <RegistrationContainer />;
      case 'topcourses':
        return <TopCourses />;
    }
  };
  
  render() {
    const { isVisible, isCovered } = this.props;
    return (
      <header className={`header ${isCovered ? 'covered' : ''}`}>
        <HeaderTitleContainer {...this.props} />
        <News />
        {this.renderSwitch(isVisible)}
      </header>
    );
  }
}
const mapStateToProps = (state) => {
  return {
    isVisible: state.isVisible,
    isCovered: state.isCovered
  }
};
const HeaderContainer = connect(mapStateToProps, null)(Header);

export default HeaderContainer;