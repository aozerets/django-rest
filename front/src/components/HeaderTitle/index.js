import React from 'react';
import './HeaderTitle.scss';
import '../../main.scss';
import { togglePage, toggleHeader, toggleLessons, togglePrograms, toggleGradebook } from '../../actions'
import {connect} from "react-redux";
import { Link } from "react-router-dom";

class HeaderTitle extends React.Component {
  constructor(props){
    super(props);
  }
  
  render() {
    console.log(this.props);
    const { isVisible, isCovered, programsOpen, lessonsOpen, gradebookOpen, togglePage, toggleHeader, togglePrograms, toggleLessons, toggleGradebook, status } = this.props;
    return (
      <div className="header__main">
        <div className="header__container flex-grid">
          <div className="header__company-logo" onClick={() => toggleHeader()} />
          <div className="header__item">
            <div onClick={() => togglePrograms(programsOpen)}><Link className="header__link" to="/programs">Programs{String.fromCharCode(9661)}</Link></div>
          </div>
          <div className="header__item">
            <div onClick={() => toggleLessons(lessonsOpen)}><Link className="header__link" to="/lessons">Lessons{String.fromCharCode(9661)}</Link></div>
          </div>
          <div className="header__item">
            Exercises
          </div>
          { status === 'teacher'
            ?
            <div className="header__item">
              <div onClick={() => toggleGradebook(gradebookOpen)}><Link className="header__link" to="/gradebook">Gradebook{String.fromCharCode(9661)}</Link></div>
            </div>
            : ''
          }
        </div>
        <div className="header__button" id="header__profile-button" onClick={() => isVisible !== 'profile' ? togglePage('profile') : togglePage('topcourses')}>
          Profile
        </div>
        <div className="header__button" id="header__login-button" onClick={() => isVisible !== 'login' ? togglePage('login') : togglePage('topcourses')}>
          Login
        </div>
        <div className="header__button" id="header__registration-button" onClick={() => isVisible !== 'registration' ? togglePage('registration') : togglePage('topcourses')}>
          SignUp
        </div>
      </div>
    );
  }
}
const mapStateToProps = (state) => {
  return {
    isVisible: state.isVisible,
    programsOpen: state.programsOpen,
    lessonsOpen: state.lessonsOpen,
    gradebookOpen: state.gradebookOpen,
    status: 'status' in state.profile ? state.profile.status : 'student'
  }
};

const mapDispatchToProps = dispatch => {
  return {
    togglePage: (page) => dispatch(togglePage(page)),
    toggleHeader: () => dispatch(toggleHeader()),
    togglePrograms: (status) => dispatch(togglePrograms(status)),
    toggleLessons: (status) => dispatch(toggleLessons(status)),
    toggleGradebook: (status) => dispatch(toggleGradebook(status))
  };
}
const HeaderTitleContainer = connect(mapStateToProps, mapDispatchToProps)(HeaderTitle);

export default HeaderTitleContainer;