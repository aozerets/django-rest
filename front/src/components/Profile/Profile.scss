[type="file"] {
  height: 0;
  overflow: hidden;
  width: 0;
}

[type="file"] + label {
  position: relative;
  transition: all 0.3s;
  vertical-align: middle;
  cursor: pointer;

  &.avatar {
    top: 1em;
    left: 30%;
    width: 40%;
    height: 1.2em;
    font-size: medium;
    border: 1px;
    box-shadow: 0 0 8px 1px #F5F5F5;
    opacity: .8;
    -webkit-transform: scale(1.4);
    transform: scale(1.4);
    font-family: 'Roboto', 'Poppins', sans-serif;
    background-color: #C864FF;
    border-radius: 50px;
    overflow: hidden;
    &:hover {
      background-color: darken(#C864FF, 30%);
    }
  }
}
.profile {
  color: #F5F5F5;
  text-align: center;
  &__flexible {
    display: flex;
  }
  &__blocked {
    display: flex;
    flex-direction: column;
    margin: 0 auto;
  }
  &__form {
    display: flex;
    flex-direction: column;
    width: 30em;
    margin: 1em auto;
    justify-content: center;
  }
  &__item {
    display: flex;
    justify-content: space-between;
    margin: .3em .3em;
    text-align: center;
  }
  &__avatar {
    margin: 2em auto;
    width: 10em;
  }
  &__date--input {
    justify-content: center;
    cursor: pointer;
    background: #C864FF;
    border-radius: 1.2rem;
    color: #fff;
    display: flex;
    padding: 0;
    margin: 1em auto;
    text-align: center;
    font-family: 'Roboto', 'Poppins', sans-serif;
    width: 40%;
    height: 1.2em;
    border: 1px;
    box-shadow: 0 0 8px 1px #F5F5F5;
    opacity: .8;
    -webkit-transform: scale(1.4);
    transform: scale(1.4);
    &:hover {
      background-color: darken(#C864FF, 30%);
    }
  }
  &__error--server {
    margin: 0 1em;
    text-align: center;
    word-break: break-word;
    color: red;
  }
  &__hide {
    display: none;
  }
  h2 {
    &__error {
      text-align: center;
      color: red;
      font-weight: bold;
    }
  }
}