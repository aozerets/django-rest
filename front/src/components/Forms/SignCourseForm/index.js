import React from 'react';
import '../../../main.scss';
import './SignCourseForm.scss';

class SignCourseForm extends React.Component {
  constructor(props) {
    super(props);
  };
  
  handleSignCourse = () => {
    this.props.handleToggle();
    alert("Congratulations!!!! U R signed");
    //event.preventDefault();
  };
  
  render() {
    const { isOpen } = this.props;
    return (
      <div id="myModal" className={isOpen ? "modal__show" : "modal"}>
        <div className="modal__content">
          <div className="modal__header">
            <span className="modal__close" id="close" onClick={this.handleSignCourse}>&times;</span>
            <h2 className="modal__title">Signing on some course</h2>
          </div>
          <div className="modal__body">
            <p className="modal__body">Give me all your money!! =)</p>
            <p className="modal__body">Some other text...</p>
          </div>
          <div className="modal__footer">
            <h3 className="modal__title">Very good choice</h3>
          </div>
        </div>
      </div>
    );
  }
}
export default SignCourseForm;