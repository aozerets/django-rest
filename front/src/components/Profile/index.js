import React from 'react';
import DatePicker from "react-datepicker";
import { getCookie2, handleErrors } from "../Utils";
import '../../main.scss';
import './Profile.scss';
import "react-datepicker/dist/react-datepicker.css";

class Profile extends React.Component {
  constructor(props) {
    super(props);
    
    this.state = ({
      username: '',
      name: '',
      surname: '',
      country: '',
      city: '',
      phone: '',
      company: '',
      position: '',
      birthDate: '',
      student_programs: '',
      user_avatar: ''
    })
  }
  
  handleProfile = ev => {
    ev.preventDefault();
    const formData = new FormData(ev.currentTarget);
    fetch('/api/v1/profile/', {
      method: 'PUT',
      headers: {'X-CSRFToken': getCookie2()},
      body: formData
    }).then(handleErrors)
      .then(() => this.props.togglePage('topcourses'))
  };
  
  handleChange = ev => this.setState({ [ev.target.name]: ev.target.value });
  
  handleBirth = date => this.setState({ birthDate: date });
  
  goToLessons = () => {
    console.log("switching");
  };
  
  componentDidMount() {
    fetch('/api/v1/profile/', {
      method: 'GET',
      headers: {'X-CSRFToken': getCookie2()},
    }).then(handleErrors)
      .then(res => res.json())
      .then(res => {
        const newState = {};
        Object.keys(res).map((key) => (res[key] == null) ? newState[key] = '' : newState[key] = res[key] );
        if (newState['birth_date']) {
          newState['birthDate'] = new Date(newState['birth_date'])
        }
        this.setState({...newState});
      })
      .catch((e) => {
        console.log(e);
      });
  }
  
  dateInput = ({onClick}) => (
    <input className="profile__date--input" onClick={onClick}>
    </input>
  );
  
  render() {
    const { name, surname, country, city, phone, company, position, birthDate, user_avatar} = this.state;
    return (
      <div className="profile">
        <h1>Personal Account</h1>
        <h2 id="profileError" className="profile__error--server" hidden>Wrong!</h2>
        <form className="profile__form" onSubmit={this.handleProfile} name="loginForm" encType="multipart/form-data">
          <div className="profile__flexible">
            <div>
              <div className="profile__item">
                <label htmlFor="name">Name: </label><input className="profile__item" type="text" name="name" placeholder="name" value={name} onChange={this.handleChange} />
              </div>
              <h4 className="profile__error" id="nameError" hidden />
              <div className="profile__item">
                <label htmlFor="surname">Surname: </label><input className="profile__item" type="text" name="surname" placeholder="surname" value={surname} onChange={this.handleChange} />
              </div>
              <h4 className="profile__error" id="surnameError" hidden />
              <div className="profile__item">
                <label htmlFor="country">Country: </label><input className="profile__item" type="text" name="country" placeholder="country" value={country} onChange={this.handleChange} />
              </div>
              <h4 className="profile__error" id="countryError" hidden />
              <div className="profile__item">
                <label htmlFor="city">City: </label><input className="profile__item" type="text" name="city" placeholder="city" value={city} onChange={this.handleChange} />
              </div>
              <h4 className="profile__error" id="cityError" hidden />
              <div className="profile__item">
                <label htmlFor="phone">Phone: </label><input className="profile__item" type="text" name="phone" placeholder="phone" value={phone} onChange={this.handleChange} />
              </div>
              <h4 className="profile__error" id="phoneError" hidden />
              <div className="profile__item">
                <label htmlFor="company">Company: </label><input className="profile__item" type="text" name="company" placeholder="company" value={company} onChange={this.handleChange} />
              </div>
              <h4 className="profile__error" id="companyError" hidden />
              <div className="profile__item">
                <label htmlFor="position">Position: </label><input className="profile__item" type="text" name="position" placeholder="position" value={position} onChange={this.handleChange} />
              </div>
              <h4 className="profile__error" id="positionError" hidden />
            </div>
            <div className="profile__blocked">
              <label htmlFor="birth_date">Date of Birth: </label>
              <DatePicker
                placeholderText="select..."
                dateFormat="yyyy-MM-dd"
                name='birth_date'
                selected={birthDate}
                customInput={this.dateInput(this)}
                onChange={date => this.handleBirth(date) }/>
              <label>Set avatar: </label>
              <input type="file" name="user_avatar" id="user_avatar"/>
              <label className="avatar" htmlFor="user_avatar">choose... </label>
              {user_avatar !== '' ? <img className='profile__avatar' src={ require('../../public/image' + user_avatar) } /> : ''}
            </div>
          </div>
          <button className="btnAction" type="submit" id="btnProfile">Confirm</button>
        </form>
      </div>
    );
  }
}

export default Profile;