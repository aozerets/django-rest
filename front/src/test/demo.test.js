import React from 'react';
import { CourseCard } from '../components/CourseCard/CourseCard';
import { Profile } from '../components/Profile/';
import { SignCourseForm } from '../components/Forms/SignCourseForm/';
import DatePicker from "react-datepicker";
import { shallow, mount } from 'enzyme';
import renderer from 'react-test-renderer';

it("adds correctly", () => {
  expect(1+1).toEqual(2);
});


describe('CourseCard rendering', () => {
  it('snapshot matching', () =>{
    const component = renderer.create(<CourseCard />);
    expect(component.toJSON()).toMatchSnapshot();
  });
  it('should render elements with right count', () => {
    let wrapper = shallow(<CourseCard />);
    expect(wrapper.children('h2')).toHaveLength(1);
    expect(wrapper.children('h4')).toHaveLength(1);
    expect(wrapper.children('p')).toHaveLength(3);
  });
  it('should highlight', () => {
    let wrapper = shallow(<CourseCard />);
    const highlight = jest.fn(() => {console.log("Punished")});
    wrapper.instance().highlightOption = highlight;
    wrapper.instance().forceUpdate();
    wrapper.find('div').simulate('mouseEnter');
    expect(highlight).toHaveBeenCalled();
    wrapper.find('div').simulate('mouseLeave');
    expect(highlight).toHaveBeenCalled();
  });
  it('should be highlight class', () => {
    let wrapper = mount(<CourseCard />);
    wrapper.find('div').simulate('mouseEnter');
    expect(wrapper.getDOMNode().classList.contains("highlighted")).toBeTruthy();
    wrapper.find('div').simulate('mouseLeave');
    expect(wrapper.getDOMNode().classList.contains("highlighted")).toBeFalsy();
  });
  it('should render with props', () => {
    const name = 'NAME';
    const title = 'TITLE';
    const started = '12.12.2012';
    let wrapper = mount(<CourseCard name={name} title={title} started={started} more={'MORE MORE'} />);
    
    expect(wrapper.find('h2').text()).toEqual(title);
    expect(wrapper.find('h4').text()).toEqual(`Starting ${started}!!!!`);
    expect(wrapper.getDOMNode().classList.contains(`course-card__${name}`)).toBeTruthy();
  })
});


describe('Profile rendering', () => {
  it('snapshot matching', () => {
    const getProfile = jest.fn();
    const component = renderer.create(<Profile getProfile={getProfile}/>);
    expect(component.toJSON()).toMatchSnapshot();
  });
  it('should render elements with right count', () => {
    const getProfile = jest.fn();
    let wrapper = shallow(<Profile getProfile={getProfile}/>);
    
    expect(wrapper.children('h1')).toHaveLength(1);
    expect(wrapper.children('h2')).toHaveLength(1);
    expect(wrapper.children('form')).toHaveLength(1);
    expect(wrapper.containsMatchingElement(<DatePicker />)).toBeTruthy();
  });
  it('should render with props', () => {
    const name = 'NAME';
    const surname = 'SURNAME';
    const country = 'COUNTRY';
    const props = {
      "name": name,
      "surname": surname,
      "country": country
    };
    let wrapper = mount(<Profile getProfile={jest.fn()} />);
    wrapper.setState({...props});
    
    expect(wrapper.find('[name="name"]').props().value).toEqual(name);
    expect(wrapper.find('[name="surname"]').props().value).toEqual(surname);
    expect(wrapper.find('[name="country"]').props().value).toEqual(country);
  })
});


describe('SignCourseForm rendering', () => {
  it('snapshot matching', () => {
    const toggleSignCourse = jest.fn();
    const component = renderer.create(<SignCourseForm isSignCourseOpen={true} toggleSignCourse={toggleSignCourse}/>);
    expect(component.toJSON()).toMatchSnapshot();
  });
  it('should render elements with right count', () => {
    let wrapper = shallow(<SignCourseForm />);

    expect(wrapper.find('h2')).toHaveLength(1);
    expect(wrapper.find('h3')).toHaveLength(1);
    expect(wrapper.find('span')).toHaveLength(1);
  });
  it('should render with props', () => {
    const h2Text = 'Signing on some course';
    const h3Text = 'Very good choice';
    let wrapper = mount(<SignCourseForm isSignCourseOpen={true}/>);
    
    expect(wrapper.find('h2').text()).toEqual(h2Text);
    expect(wrapper.find('h3').text()).toEqual(h3Text);
  });
  it('should call handleSignCourse on close', () => {
    let wrapper = shallow(<SignCourseForm isSignCourseOpen={true}/>);
    const handleSignCourse = jest.fn();
    wrapper.instance().handleSignCourse = handleSignCourse;
    wrapper.instance().forceUpdate();
    wrapper.find('span').simulate('click');
    expect(handleSignCourse).toHaveBeenCalled();
  });
  it('should call toggleSignCourse on close', () => {
    const toggleSignCourse = jest.fn(() => {console.log("Punished")});
    let wrapper = shallow(<SignCourseForm isSignCourseOpen={true} toggleSignCourse={toggleSignCourse}/>);
    wrapper.find('span').simulate('click');
    expect(toggleSignCourse).toHaveBeenCalled();
  });
});