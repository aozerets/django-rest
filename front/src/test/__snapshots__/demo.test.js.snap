// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CourseCard rendering snapshot matching 1`] = `
<div
  className="course-card course-card__undefined animate-card-in option"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <h2
    className="course-card__title"
  />
  <h4
    className="course-card__more-info"
  >
    Starting 
    !!!!
  </h4>
  <p
    className="course-card__more-info"
  >
    Some more info about this awesome course you are considering!: 
  </p>
  <p
    className="course-card__action"
  >
    <a
      className="course-card__btn"
    >
      Sign it
    </a>
  </p>
  <p
    className="course-card__action"
  >
    <a
      className="course-card__btn"
    >
      Lessons
    </a>
  </p>
</div>
`;

exports[`Profile rendering snapshot matching 1`] = `
<div
  className="profile"
>
  <h1>
    Personal Account
  </h1>
  <h2
    className="profile__error--server"
    hidden={true}
    id="profileError"
  >
    Wrong!
  </h2>
  <form
    className="profile__form"
    encType="multipart/form-data"
    name="loginForm"
    onSubmit={[Function]}
  >
    <div
      className="profile__flexible"
    >
      <div>
        <div
          className="profile__item"
        >
          <label
            htmlFor="name"
          >
            Name: 
          </label>
          <input
            className="profile__item"
            name="name"
            onChange={[Function]}
            placeholder="name"
            type="text"
            value=""
          />
        </div>
        <h4
          className="profile__error"
          hidden={true}
          id="nameError"
        />
        <div
          className="profile__item"
        >
          <label
            htmlFor="surname"
          >
            Surname: 
          </label>
          <input
            className="profile__item"
            name="surname"
            onChange={[Function]}
            placeholder="surname"
            type="text"
            value=""
          />
        </div>
        <h4
          className="profile__error"
          hidden={true}
          id="surnameError"
        />
        <div
          className="profile__item"
        >
          <label
            htmlFor="country"
          >
            Country: 
          </label>
          <input
            className="profile__item"
            name="country"
            onChange={[Function]}
            placeholder="country"
            type="text"
            value=""
          />
        </div>
        <h4
          className="profile__error"
          hidden={true}
          id="countryError"
        />
        <div
          className="profile__item"
        >
          <label
            htmlFor="city"
          >
            City: 
          </label>
          <input
            className="profile__item"
            name="city"
            onChange={[Function]}
            placeholder="city"
            type="text"
            value=""
          />
        </div>
        <h4
          className="profile__error"
          hidden={true}
          id="cityError"
        />
        <div
          className="profile__item"
        >
          <label
            htmlFor="phone"
          >
            Phone: 
          </label>
          <input
            className="profile__item"
            name="phone"
            onChange={[Function]}
            placeholder="phone"
            type="text"
            value=""
          />
        </div>
        <h4
          className="profile__error"
          hidden={true}
          id="phoneError"
        />
        <div
          className="profile__item"
        >
          <label
            htmlFor="company"
          >
            Company: 
          </label>
          <input
            className="profile__item"
            name="company"
            onChange={[Function]}
            placeholder="company"
            type="text"
            value=""
          />
        </div>
        <h4
          className="profile__error"
          hidden={true}
          id="companyError"
        />
        <div
          className="profile__item"
        >
          <label
            htmlFor="position"
          >
            Position: 
          </label>
          <input
            className="profile__item"
            name="position"
            onChange={[Function]}
            placeholder="position"
            type="text"
            value=""
          />
        </div>
        <h4
          className="profile__error"
          hidden={true}
          id="positionError"
        />
      </div>
      <div
        className="profile__blocked"
      >
        <label
          htmlFor="birth_date"
        >
          Date of Birth: 
        </label>
        <div
          className="react-datepicker-wrapper"
        >
          <div
            className="react-datepicker__input-container"
          >
            <input
              className="profile__date--input"
              disabled={false}
              name="birth_date"
              onBlur={[Function]}
              onChange={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              placeholder="select..."
              readOnly={false}
              value=""
            />
          </div>
        </div>
        <label>
          Set avatar: 
        </label>
        <input
          id="user_avatar"
          name="user_avatar"
          type="file"
        />
        <label
          className="avatar"
          htmlFor="user_avatar"
        >
          choose... 
        </label>
        
      </div>
    </div>
    <button
      className="btnAction"
      id="btnProfile"
      type="submit"
    >
      Confirm
    </button>
  </form>
</div>
`;

exports[`SignCourseForm rendering snapshot matching 1`] = `
<div
  className="modal__show"
  id="myModal"
>
  <div
    className="modal__content"
  >
    <div
      className="modal__header"
    >
      <span
        className="modal__close"
        id="close"
        onClick={[Function]}
      >
        Ã—
      </span>
      <h2
        className="modal__title"
      >
        Signing on some course
      </h2>
    </div>
    <div
      className="modal__body"
    >
      <p
        className="modal__body"
      >
        Give me all your money!! =)
      </p>
      <p
        className="modal__body"
      >
        Some other text...
      </p>
    </div>
    <div
      className="modal__footer"
    >
      <h3
        className="modal__title"
      >
        Very good choice
      </h3>
    </div>
  </div>
</div>
`;
